src/data/polynomial/eval.lean

```diff
@@ -471,12 +471,24 @@ end
 
 @[simp] lemma map_one : (1 : polynomial R).map f = 1 := eval₂_one _ _
 
+@[simp] lemma map_mul : (p * q).map f = p.map f * q.map f :=
+by { rw [map, eval₂_mul_noncomm], exact λ k, (commute_X _).symm }
+
+@[simp] lemma map_smul (r : R) : (r • p).map f = f r • p.map f :=
+by rw [map, eval₂_smul, ring_hom.comp_apply, C_mul']
+
+/-- `polynomial.map` as a `ring_hom` -/
+def map_ring_hom (f : R →+* S) : polynomial R →+* polynomial S :=
+{ to_fun := polynomial.map f,
+  map_add' := λ _ _, map_add f,
+  map_zero' := map_zero f,
+  map_mul' := λ _ _, map_mul f,
+  map_one' := map_one f }
+
+@[simp] lemma coe_map_ring_hom (f : R →+* S) : ⇑(map_ring_hom f) = map f := rfl
```

Comment:
Can `@[simps]` generate this for us?

Reply:
`@[simps? {fully_applied := ff}]` generates the right lemma but with a worse name. `simps` now knows enough to be taught how to handle a `coe_` prefix, but I don't think anyone has taught it to do so for bundled homs, so that feels like a task for a future PR.